src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -165,6 +186,29 @@ p5.RendererGL.prototype.vertex = function(x, y) {
   return this;
 };
 
+p5.RendererGL.prototype.setAttribute = function(attributeName, data){
+  // if attributeName is in one of default, throw some warning
+  if(!this._useUserAttributes){
+    this._useUserAttributes = true;
+    this.userAttributes = {};
+  }
+  const size = data.length ? data.length : 1;
+  if (!this.userAttributes.hasOwnProperty(attributeName)){
+    this.tessyVertexSize += size;
+  }
+  this.userAttributes[attributeName] = data;
+  const buff = attributeName.concat('Buffer');
+  const bufferExists = this.immediateMode
+    .buffers
+    .user
+    .some(buffer => buffer.dst === buff);
```

Comment:
No action needed here but just double check my assumptions: At first I was a tad worried that we're doing this loop for every vertex every frame, but I guess it only becomes a problem if you have a bunch of user attributes right? If you have one or two this is more than enough. Potentially if we wanted to scale this to more, we could refactor the buffers to be an object instead of an array, but that's maybe a bigger refactor than we need right now.

Reply:
Yeah it seems to be okay for the moment, you can set quite a few attributes without it slowing down noticeably, but in the future we could look into that refactor.